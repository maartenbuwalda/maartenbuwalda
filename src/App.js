import React, { Component } from 'react';
import logo from './logo.svg';
import styles from './App.sass';

class App extends Component {
  render() {
    return (
      <div className={styles["App"]}></div>
    );
  }
}

export default App;
