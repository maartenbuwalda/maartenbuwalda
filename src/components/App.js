import React, { Component } from 'react'
import Header from './Header'
import image from '../static/images/ireland2.jpg'

class App extends Component {
  componentDidMount () {
    this.props.getDominantColor(image)
  }
  render () {
    const { color, setDominantColor } = this.props
    return (
      <React.Fragment>
        <Header image={image} color={color} setDominantColor={setDominantColor}/>
      </React.Fragment>
    )
  }
}

export default App
