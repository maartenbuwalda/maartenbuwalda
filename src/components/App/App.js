import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { HashRouter } from 'react-router-dom'
import BusinessCard from '../BusinessCard'
// import Timeline from '../Timeline'
import Heading from '../../containers/Heading'
import Background from '../../containers/Background'
import Grid from '../Grid'

const PageWrapper = styled.article`
  display: flex;
  justify-content: center;
`

class App extends Component {
  render () {
    return (
      <HashRouter>
        <React.Fragment>
          <Heading />
          <PageWrapper>
            <Grid>
              <BusinessCard position="s1" />
              {/* <Timeline position="m1" /> */}
            </Grid>
          </PageWrapper>
          <Background/>
        </React.Fragment>
      </HashRouter>
    )
  }
}

App.propTypes = {
  getDominantColor: PropTypes.func,
  color: PropTypes.object,
}

export default App
