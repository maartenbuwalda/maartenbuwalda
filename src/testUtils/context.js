const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
