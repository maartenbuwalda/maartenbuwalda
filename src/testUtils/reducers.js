export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
