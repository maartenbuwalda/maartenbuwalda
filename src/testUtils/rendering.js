import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
