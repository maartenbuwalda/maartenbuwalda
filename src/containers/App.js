import App from '../components/App'
import { connect } from 'react-redux'

export const mapStateToProps = state => {
  return {
    color: state.color,
  }
}

export default connect(mapStateToProps)(App)
