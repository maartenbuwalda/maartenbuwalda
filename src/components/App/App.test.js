import React from 'react'
import App from './index'
import { componentRenders } from '../../testUtils/rendering'

jest.mock('../BusinessCard', () => 'BusinessCard')
jest.mock('../Navigation', () => 'Navigation')
jest.mock('../Pages', () => 'Pages')

describe('App component', () => {
  const component = <App />
  componentRenders(component)
})
