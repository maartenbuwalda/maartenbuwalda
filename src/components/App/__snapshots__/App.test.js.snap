// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<HashRouter>
  <ContextProvider
    value={
      Object {
        "theme": Object {
          "hex": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
          "rgb": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
        },
        "updateTheme": [Function],
      }
    }
  >
    <Background>
      <Grid>
        <Sticky>
          <BusinessCard
            position="s1"
          />
          <Navigation
            position="m1"
          />
        </Sticky>
        <Pages
          position="l1"
        />
      </Grid>
    </Background>
  </ContextProvider>
</HashRouter>
`;
